src/main/java/com/datapipe/jenkins/vault/credentials/common/VaultHelper.java

```diff
@@ -87,30 +86,25 @@ private static class SecretRetrieve extends SlaveToMasterCallable<Map<String, St
         private final String namespace;
         @CheckForNull
         private Integer engineVersion;
-        @NonNull
-        private transient ItemGroup<Item> credentialContext;
 
-        SecretRetrieve(String secretPath, String prefixPath, String namespace, Integer engineVersion, @NonNull ItemGroup<Item> credentialContext) {
+        SecretRetrieve(String secretPath, String prefixPath, String namespace, Integer engineVersion) {
             this.secretPath = secretPath;
             this.prefixPath = Util.fixEmptyAndTrim(prefixPath);
             this.namespace = Util.fixEmptyAndTrim(namespace);
             this.engineVersion = engineVersion;
-            this.credentialContext = credentialContext;
         }
 
         @Override
         public Map<String, String> call() throws IOException {
-            if (credentialContext == null) {
-                throw new IllegalStateException("The vault credential need a context");
-            }
+
 
             VaultConfiguration configuration = null;
             for (VaultConfigResolver resolver : ExtensionList.lookup(VaultConfigResolver.class)) {
                 if (configuration != null) {
                     configuration = configuration
-                        .mergeWithParent(resolver.getVaultConfig(credentialContext));
+                    .mergeWithParent(resolver.getVaultConfig((ItemGroup)Jenkins.getInstanceOrNull()));
```

Comment:
```suggestion
                    .mergeWithParent(resolver.getVaultConfig((ItemGroup)Jenkins.get()));
```

Get should be non null.